CipherText/Rotator/Rotator.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: jjose00
+ * Date: 11/7/17
+ * Time: 8:11 AM
+ */
+
+namespace Giftcards\Encryption\CipherText\Rotator;
+
+use Giftcards\Encryption\CipherText\Rotator\Store\StoreRegistry;
+use Giftcards\Encryption\Encryptor;
+
+class Rotator
+{
+    /**
+     * @var Encryptor
+     */
+    private $encryptor;
+    /**
+     * @var StoreRegistry
+     */
+    private $storeRegistry;
+
+    public function __construct(Encryptor $encryptor, StoreRegistry $storeRegistry)
+    {
+        $this->encryptor = $encryptor;
+        $this->storeRegistry = $storeRegistry;
+    }
+
+    public function rotate($storeName, $newProfile, Bounds $bounds, RotatorObserverInterface $observer)
+    {
+        $store = $this->storeRegistry->get($storeName);
+        /** @var int $offset */
+        /** @var int $limit */
+        foreach ($bounds as $key => $entry) {
+            list($offset, $limit) = $entry;
+            $records = $store->fetch($offset, $limit);
+            $observer->fetchedRecords($offset, $limit, $records);
```

Comment:
im thinking this can be moved below the if. If no record were fetched it doesnt seem correct to tell the observer that records were fetched